pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -1173,12 +1355,19 @@ impl Timeline {
             if &last_key == key {
                 accumulated_values.push(item);
             } else {
-                let (deltas, image) =
-                    flush_accumulated_states(self, last_key, &accumulated_values, gc_cutoff)
-                        .await?;
+                let retention = self
+                    .generate_key_retention(
+                        last_key,
+                        &accumulated_values,
```

Comment:
This would require the function to take `history: &mut Vec<&(Key, Lsn, Value)>` instead of `history: &[&(Key, Lsn, Value)]`, and it makes writing tests harder b/c we need to generate a history every time if we reuse some part of the history. This function is only used twice in the non-test code and I feel it's not too hard to track down where we clear the values.